client/finality-grandpa/src/observer.rs

```diff
@@ -422,12 +432,17 @@ mod tests {
 		).unwrap();
 
 		let (_tx, voter_command_rx) = tracing_unbounded("");
+
+		let finality_notifiers = Arc::new(parking_lot::Mutex::new(vec![]));
+		let justification_sender = GrandpaJustificationSender::new(finality_notifiers.clone());
```

Comment:
Why not have a dummy implementation instead? (For instance `GrandpaJustuficationSender` can be an enum)

Reply:
Or maybe just take `Option<GrandpaJustificationSender>` in environment.